src/libs/Navigation/Navigation.js

```diff
@@ -70,6 +70,24 @@ const getActiveRouteIndex = function (route, index) {
     return index;
 };
 
+function getPathsForRoutesInRootNavigator() {
+    let currentState = navigationRef.getRootState();
+    const paths = [];
+
+    while (currentState.routes.length > 0) {
+        try {
+            const path = getPathFromState(currentState, linkingConfig.config);
+            if (path) {
+                paths.push(path.substring(1));
+            }
+        } catch (e) {
+            return paths;
+        }
+        currentState = {...currentState, routes: currentState.routes.slice(0, -1), index: currentState.index - 1};
+    }
+    return paths;
+}
```

Comment:
I think this may cause a performance degradation for users who keep using the app for long periods. The more you use the app the more paths we have to extract. Most of the paths we get will be a waste. I to implement an early return mechanism.

```js
function getIndexOfPathInRootNavigator(path) {
    let currentState = navigationRef.getRootState();

    for (let index = 0; index < 5; index++) {
        if (!currentState.routes.length) {
            break;
        }

        const pathFromState = getPathFromState(currentState, linkingConfig.config);
        if (path === pathFromState.substring(1)) {
            return index;
        }

        currentState = {...currentState, routes: currentState.routes.slice(0, -1), index: currentState.index - 1};
    }

    return -1;
} 
```



The 5 limit is used as a threshold. There would be no case where we would pop more than 4 screens. Currently I think the max to pop is 2. So maybe drop the 5 to 3.